client/my-sites/plans-grid/components/features-grid/index.tsx

```diff
@@ -400,13 +400,15 @@ class FeaturesGrid extends Component< FeaturesGridType > {
 						canUserPurchasePlan={ canUserPurchasePlan }
 						availableForPurchase={ availableForPurchase }
 						className={ getPlanClass( planSlug ) }
-						trialPlan={ isFreeHostingTrial( planSlug ) }
 						freePlan={ isFreePlan( planSlug ) }
+						freeTrialPlanSlug={ freeTrialPlanSlug }
 						isWpcomEnterpriseGridPlan={ isWpcomEnterpriseGridPlan( planSlug ) }
 						isWooExpressPlusPlan={ isWooExpressPlusPlan( planSlug ) }
 						isInSignup={ isInSignup }
 						isLaunchPage={ isLaunchPage }
-						onUpgradeClick={ () => handleUpgradeClick( planSlug ) }
+						onUpgradeClick={ ( overridePlanSlug ) =>
```

Comment:
We can wrap this with useCallback to pass same reference on rerenders. I know it wasn't done before, so totally fine to ignore.

Reply:
I can't do this here, the plan slug from the mapped data is used as a dependency, and this is not a functional component.